src/terminal/terminal/terminal.h

```diff
@@ -915,6 +509,75 @@ void guac_terminal_scroll_handler(guac_terminal_scrollbar* scrollbar, int value)
 void guac_terminal_dup(guac_terminal* term, guac_user* user,
         guac_socket* socket);
 
+/**
+ * Returns the number of rows within the buffer of the given terminal which are
+ * not currently displayed on screen. Adjustments to the desired scrollback
+ * size are taken into account, and rows which are within the buffer but
+ * unavailable due to being outside the desired scrollback range are ignored.
+ *
+ * @param term
+ *     The terminal whose off-screen row count should be determined.
+ *
+ * @return
+ *     The number of rows within the buffer which are not currently displayed
+ *     on screen.
+ */
+int guac_terminal_available_scroll(guac_terminal* term);
+
+/**
+ * Returns the height of the given terminal, in characters.
+ *
+ * @param term
+ *     The terminal whose height should be determined.
+ *
+ * @return
+ *     The height of the terminal, in characters.
+ */
+int guac_terminal_term_height(guac_terminal* term);
+
+/**
+ * Returns the width of the given terminal, in characters.
+ *
+ * @param term
+ *     The terminal whose width should be determined.
+ *
+ * @return
+ *     The width of the terminal, in characters.
+ */
+int guac_terminal_term_width(guac_terminal* term);
+
+/**
+ * Clears the clipboard contents for a given terminal, and assigns a new
+ * mimetype for future data.
+ *
+ * @param terminal The terminal whose clipboard is being reset.
+ * @param mimetype The mimetype of future data.
```

Comment:
Please migrate to the newer style we use for `@param`, `@return`, etc. For example:

https://github.com/apache/guacamole-server/blob/6dd33a8d90907b93a29bc482e888e4a972a0be4d/src/terminal/terminal/terminal.h#L874-L878